### ISSUE TITLE
How to apply SHAP to Multiple Regression Problem

### ISSUE THREAD

**Issue by coryroyce** [shap/shap#1815]
I'm trying to apply SHAP to a Multiple Regression problem, and I'm not sure where to start.

Details: Using a Tensorflow Keras Layer model that has more than 75 regression outputs, and would like to use DeepExplainer, GradientExplainer, or KernelExplainer.

 Below is some sample code based on [Deep Learning Models for Multi-Output Regression](https://machinelearningmastery.com/deep-learning-models-for-multi-output-regression/)

Any feedback or reference material would be appreciated.

```
# use mlp for prediction on multi-output regression
from numpy import asarray
from sklearn.datasets import make_regression
from keras.models import Sequential
from keras.layers import Dense
 
# get the dataset
def get_dataset():
	X, y = make_regression(n_samples=1000, n_features=10, n_informative=5, n_targets=75, random_state=2)
	return X, y
 
# get the model
def get_model(n_inputs, n_outputs):
	model = Sequential()
	model.add(Dense(20, input_dim=n_inputs, kernel_initializer='he_uniform', activation='relu'))
	model.add(Dense(n_outputs, kernel_initializer='he_uniform'))
	model.compile(loss='mae', optimizer='adam')
	return model
 
# load dataset
X, y = get_dataset()
n_inputs, n_outputs = X.shape[1], y.shape[1]
# get model
model = get_model(n_inputs, n_outputs)
# fit the model on all data
model.fit(X, y, verbose=0, epochs=100)
# make a prediction for new data
row = [-0.99859353,2.19284309,-0.42632569,-0.21043258,-1.13655612,-0.55671602,-0.63169045,-0.87625098,-0.99445578,-0.3677487]
newX = asarray([row])
yhat = model.predict(newX)
print('Predicted: %s' % yhat[0])
```

**Comment by coryroyce** [comment]
I have been working on this one for a few days now and have started getting a working solution. Where is the best location for me to create a demo .ipynb notebook and add it to the repository?

**Comment by coryroyce** [comment]
I resolved this issue by using the kernel explainer and then adding a drop down to dynamically select the label to explore with SHAP. Full code and tutorial on [github](https://github.com/coryroyce/Explainable_AI/blob/main/Notebooks/SHAP_Values_for_Multiple_Regression_Models.ipynb)

**Comment by mihailyanchev** [comment]
> I resolved this issue by using the kernel explainer and then adding a drop down to dynamically select the label to explore with SHAP. Full code and tutorial on [github](https://github.com/coryroyce/Explainable_AI/blob/main/Notebooks/SHAP_Values_for_Multiple_Regression_Models.ipynb)

Hi Cory, can you re-share this notebook again, because the link is not working? Is this the solution: https://shap.readthedocs.io/en/latest/example_notebooks/tabular_examples/model_agnostic/Multioutput%20Regression%20SHAP.html

**Comment by coryroyce** [comment]
Here is the link to the original file (I'm not sure how to fix the link in the Docs you shared above:
https://github.com/slundberg/shap/blob/master/notebooks/tabular_examples/model_agnostic/Multioutput%20Regression%20SHAP.ipynb

**Comment by junaid1990** [comment]
How we can draw scatter plot for multioutput regression? Wrapper models.
And how to visualize partial dependence plots? Please
